src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1220,10 +1220,18 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
                 throw new NotSupportedException(WorkspacesResources.Changing_document_property_is_not_supported);
             }
 
-            if (!this.CanApplyChange(ApplyChangesKind.ChangeDocument)
-                && projectChanges.GetChangedDocuments(onlyGetDocumentsWithTextChanges: true).Any())
+            foreach (var changedDocumentId in projectChanges.GetChangedDocuments(onlyGetDocumentsWithTextChanges: true))
             {
-                throw new NotSupportedException(WorkspacesResources.Changing_documents_is_not_supported);
+                if (!this.CanApplyChange(ApplyChangesKind.ChangeDocument))
```

Comment:
Since this applies to the workspace, is there any way to avoid checking this more than once?

Reply:
Sure. Fixed